src/modules/content-tree/content.tree.module.js

```diff
@@ -2,20 +2,79 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import List from './components/list/list.component';
 
+const CLASS_IS_TREE_RESIZING = 'ez-is-tree-resizing';
+
 export default class ContentTreeModule extends Component {
     constructor(props) {
         super(props);
 
+        this.changeContainerWidth = this.changeContainerWidth.bind(this);
+        this.addWidthChangeListener = this.addWidthChangeListener.bind(this);
+        this.removeWidthChangeListener = this.removeWidthChangeListener.bind(this);
+        this._refTreeContainer = React.createRef();
+
         this.state = {
             selectedLocationId: props.selectedLocationId,
             locations: props.preloadedLocations,
+            resizeStartPositionX: 0,
+            containerWidth: 0,
+            resizedContainerWidth: 0,
+            isResizing: false,
         };
     }
 
+    componentWillUnmount() {
+        this.clearDocumentResizingState();
+    }
+
+    changeContainerWidth({ clientX }) {
+        const currentPositionX = clientX;
+
+        this.setState((state) => ({
+            resizedContainerWidth: state.containerWidth + (currentPositionX - state.resizeStartPositionX),
+        }));
+    }
+
+    addWidthChangeListener({ nativeEvent }) {
+        const resizeStartPositionX = nativeEvent.clientX;
+        const containerWidth = parseInt(window.getComputedStyle(this._refTreeContainer.current).width, 10);
```

Comment:
No it's not the same when running this function for the first time. I don't want to get the width of a container when a component is mounted because it might be still not visible. By this approach you get a container width value when it's actually visible.